### PR TITLE
TOK-541: add ABI

### PR DESCRIPTION
**Pull request by antomor**
## What

- Add ABI

### CODE REVIEW

File: src/app/collective-rewards/metrics/components/ABIMetrics/hooks/useGetABI.ts [hunk 1]
```diff
@@ -1,33 +1,104 @@
...
     error: totalAllocationError,
-  } = useGetTotalAllocation(gauges ?? [])
-  // FIXME: verify if we need to use the cyclePayout in wei or in RIF and if we need to use bigint or Number
-  const rewardsPerStRIFPerCycle = Number(
-    ((cyclePayout ?? 1n) * weightedAverageBuilderRewardsPct) / totalAllocation,
+  } = useGaugesGetFunction(gauges, 'totalAllocation')
+  const sumTotalAllocation = Object.values(totalAllocation ?? {}).reduce(
+    (acc, value) => acc + (value ?? 0n),
+    1n,
   )
```
Comment (antomor) [inline src/app/collective-rewards/metrics/components/ABIMetrics/hooks/useGetABI.ts:42]: Why couldn't we use the hook `useGetTotalAllocation`?

File: src/app/collective-rewards/metrics/components/ABIMetrics/hooks/useGetABI.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+    if (!sumTotalAllocation) {
+      return 0
+    }
+
+    const rifPrice = prices.RIF?.price ?? 0
+    const rbtcPrice = prices.RBTC?.price ?? 0
+    const rifAmount = Number(formatEther(rifRewards ?? 0n))
+    const rbtcAmount = Number(formatEther(rbtcRewards ?? 0n))
+    const cyclePayout = rifAmount * rifPrice + rbtcAmount * rbtcPrice
```
Comment (franciscotobar) [inline src/app/collective-rewards/metrics/components/ABIMetrics/hooks/useGetABI.ts:66]: Im not sure if I should extract this into a hook or function, if I do it we will introduce another useMemo

File: src/app/collective-rewards/metrics/components/ABIMetrics/hooks/useGetABI.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+        (acc, { allocation, current }) =>
+          acc + Number(((allocation * 100n) / sumTotalAllocation) * BigInt(current)),
+        0,
+      ) / 100
+
+    const totalAllocationInEther = Number(formatEther(sumTotalAllocation))
+    const rewardsPerStRIFPerCycle =
+      (cyclePayout * (weightedAverageBuilderRewardsPct / totalAllocationInEther / rifPrice)) / 100
+
+    return (Math.pow(1 + rewardsPerStRIFPerCycle, 26) - 1) * 100
```
Comment (antomor) [inline src/app/collective-rewards/metrics/components/ABIMetrics/hooks/useGetABI.ts:95]: @franciscotobar why do we need to divide by 100 and then multiply by 100? I think the formula provided was:
```suggestion
    const rewardsPerStRIFPerCycle = cyclePayout * (weightedAverageBuilderRewardsPct / totalAllocationInEther)
    return Math.pow(1 + rewardsPerStRIFPerCycle / rifPrice, 26) - 1
```

File: src/app/collective-rewards/metrics/components/ABIMetrics/hooks/useGetABI.ts (hunk 2 above)
Comment (antomor) [reply src/app/collective-rewards/metrics/components/ABIMetrics/hooks/useGetABI.ts:66]: I don't think we need it for now 👍🏼

File: src/app/collective-rewards/utils/isBuilderOperational.ts [hunk 4]
```diff
@@ -17,6 +17,16 @@ export const isBuilderActive = (stateFlags?: BuilderStateFlags) => {
   )
 }
 
+export const isBuilderRewarded = (stateFlags?: BuilderStateFlags) => {
```
Comment (antomor) [inline src/app/collective-rewards/utils/isBuilderOperational.ts:20]: I don't get why this reward is linked to the state. Should we rename it to `isBuilderRewardable`?

**Review by franciscotobar — APPROVED** [review]
LGTM

**Pull request merged**